src/main/java/ogc/rs/processes/s3PreSignedURLGeneration/S3PreSignedURLGenerationProcess.java

```diff
@@ -0,0 +1,299 @@
+                        }
+
+                        // Extract resourceId name (riName) and resourceGroup
+                        requestInput.put("riName", resourceInfo.getString("name"));
+                        String resourceGroupId = resourceInfo.getString("resourceGroup");
+
+                        // Make second API call to get resourceGroup info
+                        webClient.get(catServerPort, catServerHost, catRequestUri)
+                                .addQueryParam("id", resourceGroupId).send()
+                                .onSuccess(groupResponse -> {
+                                    if (groupResponse.statusCode() == 200) {
+                                        JsonObject groupInfo = groupResponse.bodyAsJsonObject().getJsonArray("results").getJsonObject(0);
+                                        requestInput.put("rgName", groupInfo.getString("name"));
+
+                                        // Now construct objectKeyName
+                                        String fileType = requestInput.getString("fileType").toLowerCase(); // Convert to lowercase
+                                        String fileExtension = fileTypeMap.getOrDefault(fileType, "");
+                                        if (fileExtension.isEmpty()) {
+                                            promise.fail("Unsupported file type");
+                                            return;
+                                        }
+
+                                        String objectKeyName = requestInput.getString("rgName") + "/" + requestInput.getString("riName") + fileExtension;
+                                        LOGGER.debug("Object key name is: {}", objectKeyName);
+                                        requestInput.put("objectKeyName", objectKeyName);
+
+                                        // Complete with the updated requestInput
+                                        promise.complete(requestInput);
+                                    } else {
+                                        LOGGER.error("Resource group not found");
+                                        promise.fail("Resource group not found");
+                                    }
+                                }).onFailure(failureResponse -> {
+                                    LOGGER.error("Failed to fetch resource group: " + failureResponse.getMessage());
+                                    promise.fail("Failed to fetch resource group");
+                                });
+                    } else {
+                        LOGGER.error("Resource not found");
+                        promise.fail("Resource not found");
+                    }
+                }).onFailure(failureResponse -> {
+                    LOGGER.error("Failed to fetch resource: " + failureResponse.getMessage());
+                    promise.fail("Failed to fetch resource");
+                });
+
+        return promise.future();
+    }
```

Comment:
This method is doing too many things - it may be better to have:
- One generic method (e.g. `makeCatApiRequest`) that only calls the catalogue `item` API
- Another method that calls `makeCatApiRequest` to first check resource ID, do ownership check, then again call `makeCat...` to check RG ID and then return the `objectKeyName`
- Avoid the nested if-elses using `.compose` and an `onSuccess`+`onFailure` only at the end

Reply:
> This method is doing too many things - it may be better to have:
> 
>     * One generic method (e.g. `makeCatApiRequest`) that only calls the catalogue `item` API
> 
>     * Another method that calls `makeCatApiRequest` to first check resource ID, do ownership check, then again call `makeCat...` to check RG ID and then return the `objectKeyName`
> 
>     * Avoid the nested if-elses using `.compose` and an `onSuccess`+`onFailure` only at the end

https://github.com/datakaveri/ogc-resource-server/pull/190/commits/ab9735b362a6a05aee9d69f98553e9ec65818162